common/src/main/java/org/hawkular/accounts/common/TokenVerifier.java

```diff
@@ -32,11 +39,41 @@
     @Inject @RealmName
     private String realm;
 
+    @Inject @HostSynonyms
+    private Set<String> hostSynonyms;
+
     @Inject
     AuthServerRequestExecutor executor;
 
     public String verify(String token) throws Exception {
-        String tokenUrl = baseUrl
+        JWSInput jwsInput;
+        try {
+            jwsInput = new JWSInput(token);
+        } catch (Exception e) {
+            throw new VerificationException("Couldn't parse token", e);
+        }
+
+        AccessToken accessToken;
+        try {
+            accessToken = jwsInput.readJsonContent(AccessToken.class);
+        } catch (IOException e) {
+            throw new VerificationException("Couldn't parse token signature", e);
+        }
+
+        URL backendUrl = new URL(accessToken.getIssuer());
+        URL baseUrlToCall = new URL(baseUrl);
+        if (!backendUrl.getHost().equalsIgnoreCase(baseUrlToCall.getHost())) {
+            if (hostSynonyms.contains(backendUrl.getHost())) {
```

Comment:
Indeed, I don't like the idea of adding more and more options. On this case, however, I'd like to get a discussion with the Keycloak team before deciding whether implementing such measures will happen in Hawkular or in Keycloak. As this was causing some troubles during development, I think this fix is good enough while a bigger discussion happens.